### ISSUE TITLE
Throttle enhancements

### ISSUE THREAD

**Issue by evictor** [carlsverre/wraptor#2]
It would be great if `throttle` could be a little more robust:

* Return value indicating whether the function call was 'dropped' or actually went through (Bool)
* Allow for automatic calling of the function after a sleep

**Comment by carlsverre** [comment]
I think the first change would be a useful addition.  The second change is difficult to do without taking a dependency on the user's architecture.  A thread may be unacceptable in an evented system - and visa-versa.  The use case I am using Throttle for is to limit how often "cache updating" code runs in worker thread.  Since these workers are running Throttle in an infinite loop - it is guaranteed that it will eventually be run after the timeout has occurred.

If you want to submit the first change in a safe way I am ok with that.  Not sure about the second one though for the above reasons.

**Comment by evictor** [comment]
Roger. 👍

**Comment by evictor** [comment]
Having some difficulty running tests. I get this:

```
$ python setup.py test 
running test
Traceback (most recent call last):
  File "setup.py", line 84, in <module>
    cmdclass={ 'test': PyTest },
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 971, in run_command
    cmd_obj.ensure_finalized()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 109, in ensure_finalized
    self.finalize_options()
  File "setup.py", line 35, in finalize_options
    TestCommand.finalize_options(self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/setuptools/command/test.py", line 78, in finalize_options
    if self.test_runner is None:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 105, in __getattr__
    raise AttributeError, attr
AttributeError: test_runner
```

Looks like the test `if self.test_runner is None` causes this error. This is not a `wraptor` thing it appears but I'm not familiar with this error or test environment.

Any ideas?

**Comment by carlsverre** [comment]
The version of pytest that I wrote Wraptor against is 4 years out of date.  I just updated the system to work - you should be able to run `make test` and have the tests run.  It only requires virtualenv to be installed.

**Comment by carlsverre** [comment]
Closed in #3
